osu.Game/Overlays/Mods/EditPresetPopover.cs

```diff
@@ -0,0 +1,203 @@
+                                Text = "Use Current Mods",
+                                Action = trySaveCurrentMod
+                            },
+                            createButton = new ShearedButton
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Text = Resources.Localisation.Web.CommonStrings.ButtonsSave,
+                                Action = tryEditPreset
+                            },
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Body.BorderThickness = 3;
+            Body.BorderColour = colours.Orange1;
+
+            nameTextBox.Current.Value = preset.Name;
+            descriptionTextBox.Current.Value = preset.Description;
+
+            createButton.DarkerColour = colours.Orange1;
+            createButton.LighterColour = colours.Orange0;
+            createButton.TextColour = colourProvider.Background6;
+
+            selectedMods.BindValueChanged(_ => updateActiveState(), true);
+
+            scrollContent.ChildrenEnumerable = preset.Mods.Select(mod => new ModPresetRow(mod));
+        }
+
+        private void trySaveCurrentMod()
+        {
+            if (!checkCanBeSave())
+            {
+                Body.Shake();
+                return;
+            }
+
+            saveModAfterClosed = selectedMods.Value.ToList();
+            scrollContent.Clear();
+            scrollContent.ChildrenEnumerable = saveModAfterClosed.Select(mod => new ModPresetRow(mod));
+            updateActiveState();
+        }
+
+        private void updateActiveState()
+        {
+            if (checkCanBeSave())
+            {
+                useCurrentModButton.DarkerColour = colours.Blue1;
+                useCurrentModButton.LighterColour = colours.Blue0;
+                useCurrentModButton.TextColour = colourProvider.Background6;
+            }
+            else
+            {
+                useCurrentModButton.DarkerColour = colours.Blue3;
+                useCurrentModButton.LighterColour = colours.Blue4;
+                useCurrentModButton.TextColour = colourProvider.Background2;
+            }
+        }
+
+        private bool checkCanBeSave()
+        {
+            if (!selectedMods.Value.Any())
+                return false;
+
+            if (saveModAfterClosed.Any())
+            {
+                return !new HashSet<Mod>(saveModAfterClosed).SetEquals(selectedMods.Value);
+            }
+
+            return button.CheckCurrentModCanBeSave();
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            ScheduleAfterChildren(() => GetContainingInputManager().ChangeFocus(nameTextBox));
+        }
+
+        private void tryEditPreset()
```

Comment:
This is weird. If you're going to disallow saving with an empty name, make the save button disabled in that case rather than this.